chore: Flatten ComputeFirewallPolicyRule directory

gemmahou opened the pull request:
### Change description

<!--
Describe what this pull request does.

* If your pull request is to address an open issue, indicate it by specifying the
issue number: 
* If your pull request fixes an issue which has not been filed, please file the
issue and put the number here.

For example: "Fixes #858"
-->
Address https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/5191 Action 2

#### Special notes for your reviewer:

#### Does this PR add something which needs to be 'release noted'?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note

```

- [ ] Reviewer reviewed release note.

#### Additional documentation e.g., references, usage docs, etc.:

<!--
This section can be blank if this pull request does not require any additional documentation.

When adding links which point to resources within git repositories, like
usage documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

#### Intended Milestone

Please indicate the intended milestone. 
- [ ] Reviewer tagged PR with the actual milestone.

### Tests you have done

<!--

Make sure you have run "make ready-pr" to run required tests and ensure this PR is ready to review. 

Also if possible, share a bit more on the tests you have done. 

For example if you have updated the pubsubtopic sample, you can share the test logs from running the test case locally.

go test -v -tags=integration ./config/tests/samples/create -test.run TestAll -run-tests pubsubtopic

-->

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

yuwenma commented:
issue #5191

yuwenma reviewed (commented):
Suggest not introducing API changes in this PR (or explain each change if necessary)

gemmahou commented on apis/compute/v1beta1/firewallpolicyrule_types.go:
```diff
@@ -27,106 +25,112 @@ var (
 )
 
 // +kcc:proto=google.cloud.compute.v1.FirewallPolicyRuleMatcherLayer4Config
-type FirewallPolicyRuleLayer4Configs struct {
```
This naming change is necessary, so it matches the proto field name to ensure correct mapper generation

gemmahou commented:
> > Suggest not introducing API changes in this PR (or explain each change if necessary)
> 
> All of these are updated descriptions, copied from the generated_types to ensure they are up-to-date with the proto. Is this change acceptable, or should we revert it?

@yuwenma discussed offline, I have reverted description changes, and added comments to necessary field name changes.

yuwenma commented on apis/compute/v1beta1/firewallpolicyrule_types.go:
```diff
@@ -27,106 +25,112 @@ var (
 
 // +kcc:proto=google.cloud.compute.v1.FirewallPolicyRuleMatcherLayer4Config
-type FirewallPolicyRuleLayer4Configs struct {
+type FirewallPolicyRuleMatcherLayer4Config struct {
```
Same question, why do we change the field name?

yuwenma reviewed (commented):
/approve
/lgtm 


Looks great. Thanks. 

One reason I'm concerned about the manual handling is the potential bugs in nil pointer reference. Here're some problems we previously found in ComputeFirewallpolicy and ComputeForwardingRule https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/5055/commits
